packages/core/src/electron-browser/menu/electron-menu-contribution.ts

```diff
@@ -119,6 +120,31 @@ export class ElectronMenuContribution implements FrontendApplicationContribution
         });
     }
 
+    handleTitleBarStyling(app: FrontendApplication): void {
+        this.hideTopPanel(app);
+        electron.ipcRenderer.on('original-titleBarStyle', (_event, style: string) => {
+            this.titleBarStyle = style;
+            this.preferenceService.ready.then(() => {
+                this.preferenceService.set('window.titleBarStyle', this.titleBarStyle, PreferenceScope.User);
+            });
+        });
+        electron.ipcRenderer.send('request-titleBarStyle');
+        this.preferenceService.ready.then(() => {
+            this.setMenu(app);
+            electron.remote.getCurrentWindow().setMenuBarVisibility(true);
+            setTimeout(() => {
+                this.titleBarStyleChangeFlag = true;
+            }, 1000);
```

Comment:
This `setTimeout` is a little odd. What's the significance of 1 second in this context?

Reply:
I removed the timeout. It was no longer necessary. We can set the flag after the first change in preferences happens.